### PR TITLE
remove the need for a double copy to minio

### PR DESCRIPTION
**Pull request by neildsouth**
### Description

on a store request the gateway currently
saves to file locally,
saves to Minio temp folder
moved from temp folder to final destination

This branch skips the second step and saves directly to the final folder.

### Status
**Work in progress**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [ ] User guide updated.
- [ ] I have updated the changelog

### CODE REVIEW

File: src/InformaticsGateway/Test/Services/Connectors/PayloadMoveActionHandlerTest.cs [hunk 1]
```diff
@@ -190,25 +190,31 @@ public async Task GivenAPayload_WhenAllFilesAreMove_ExpectPayloadToBeAddedToNoti
             });
 
             var correlationId = Guid.NewGuid();
+            var file1 = new DicomFileStorageMetadata(correlationId.ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString());
```
Comment (lillie-dae) [inline src/InformaticsGateway/Test/Services/Connectors/PayloadMoveActionHandlerTest.cs:193]: would it be worth declaring a ctor that can take GUID?

example
image


`        
public DicomFileStorageMetadata(Guid associationId, Guid identifier, Guid studyInstanceUid, Guid seriesInstanceUid, Guid sopInstanceUid) : this(
            associationId.ToString(),
            identifier.ToString(),
            studyInstanceUid.ToString(),
            seriesInstanceUid.ToString(),
            sopInstanceUid.ToString())
        { }
`

File: src/InformaticsGateway/Services/Storage/ObjectUploadService.cs [hunk 2]
```diff
@@ -192,12 +192,14 @@ private async Task UploadFileAndConfirm(string identifier, StorageObjectMetadata
...
             do
             {
-                await UploadFile(storageObjectMetadata, source, workflows, cancellationToken).ConfigureAwait(false);
+                await UploadFile(storageObjectMetadata, source, workflows, payloadId, cancellationToken).ConfigureAwait(false);
                 if (count-- <= 0)
                 {
                     throw new FileUploadException($"Failed to upload file after retries {identifier}.");
                 }
-            } while (!(await VerifyExists(storageObjectMetadata.GetTempStoragPath(_configuration.Value.Storage.RemoteTemporaryStoragePath), cancellationToken).ConfigureAwait(false)));
+            } while (!(
```
Comment (lillie-dae) [inline src/InformaticsGateway/Services/Storage/ObjectUploadService.cs:200]: would it be better to use polly to do retries on this bit?

**Review by lillie-dae — APPROVED** [review]

**Comment by mocsharp** [comment]
This PR removes the ability to have the same instance in multiple payloads.  
There was a discussion (some time ago) to have rule-based instance grouping which could result in sending the same instance to multiple payloads.

File: src/InformaticsGateway/Test/Services/Connectors/PayloadMoveActionHandlerTest.cs (hunk 1 above)
Comment (neildsouth) [reply src/InformaticsGateway/Test/Services/Connectors/PayloadMoveActionHandlerTest.cs:193]: looking at the code the only benefit for the above change would be the tests. so ....

**Comment by JoeBatt1989** [comment]
We have ran some performance tests on this version and seen some significant improvements when sening the same data through (CT 326 slices). We are also seeing MinIO not reporting that we need to reduce our request rate which does sometimes leave the MIG is a broken state.

Previous baseline tests for MIG: 

- Avg = 34.4
- Max = 36.2

New baseline tests for MIG: 

- Avg = 14
- Max = 16.2

`2023-06-08 09:51:14.1811|712|INFO|Monai.Deploy.InformaticsGateway.Services.Connectors.PayloadNotificationActionHandler||Workflow request published to md.workflow.request, message ID=33e05f11-c2bb-4492-ab29-25fd38a2135e. Payload took 00:00:13.1329433 to complete.
2023-06-08 09:52:46.3083|712|INFO|Monai.Deploy.InformaticsGateway.Services.Connectors.PayloadNotificationActionHandler||Workflow request published to md.workflow.request, message ID=01f9c934-d8aa-49f4-8f69-4bda2894e53a. Payload took 00:00:14.0552913 to complete.
2023-06-08 09:53:23.4085|712|INFO|Monai.Deploy.InformaticsGateway.Services.Connectors.PayloadNotificationActionHandler||Workflow request published to md.workflow.request, message ID=a8fc9cf1-c441-4ed5-96c8-9f6fdd5e0207. Payload took 00:00:13.9584748 to complete.
2023-06-08 09:53:50.4880|712|INFO|Monai.Deploy.InformaticsGateway.Services.Connectors.PayloadNotificationActionHandler||Workflow request published to md.workflow.request, message ID=06f49a60-4afe-4c85-a840-bf57b709cfea. Payload took 00:00:12.7193439 to complete.
2023-06-08 09:54:12.5633|712|INFO|Monai.Deploy.InformaticsGateway.Services.Connectors.PayloadNotificationActionHandler||Workflow request published to md.workflow.request, message ID=0d65548e-921e-488e-876e-00d420cc6b7d. Payload took 00:00:16.2113721 to complete.
`

**Review by woodheadio — APPROVED** [review]

**Pull request merged**